DOCS-2626 Browser Test Edits & Permissions Update (#12919)
DataDog/documentation: content/en/synthetics/browser_tests/_index.md

<div class="alert alert-info">If you are interested in testing applications that sit behind MFA, read <a href="/synthetics/guide/app-that-requires-login/#multi-factor-authentication" target="_blank">the dedicated guide </a> and <a href="https://docs.google.com/forms/d/e/1FAIpQLSdjx8PDZ8kJ3MD2ehouTri9z_Fh7PoK90J8arRQgt7QFgFxog/viewform?usp=sf_link">send feedback</a> to the Synthetic Monitoring team to help improve the systems that matter most to your teams.</div>

## Test configuration

Define the configuration of your browser test.

1. **Starting URL**: The URL from which your browser test starts the scenario.
2. **Advanced Options** (optional): Set specific options for your browser test.

  {{< tabs >}}

  {{% tab "Request Options" %}}

  Select **Disable CORS** to prevent the cross-origin resource sharing (CORS) policy from blocking your test.

  * **Request Headers**: Define headers in the **Name** and **Value** fields to add to or override the default browser headers. For example, you can set the User Agent in the header to [identify Datadog scripts][1].

>   * **Cookies**: Define cookies to add to the default browser cookies. Set multiple cookies as `<COOKIE_NAME1>=<COOKIE_VALUE1>; <COOKIE_NAME2>=<COOKIE_VALUE2>` and enter one cookie per line.

Reviewer: is it 1 cookie per line or semicolon separated?

Author: Answered you in Slack!

>   * **Cookies**: Define cookies to add to the default browser cookies. Set multiple cookies as `<COOKIE_NAME1>=<COOKIE_VALUE1>; <COOKIE_NAME2>=<COOKIE_VALUE2>` and enter one cookie per line.

Reviewer: ```suggestion
  * **Cookies**: Define cookies to add to the default browser cookies. Enter one cookie per line, using the syntax of [Set-Cookie][2].
```

  * **HTTP Authentication**: Authenticate through HTTP Basic, Digest, or NTLM with a username and a password. Your credentials are used in every step of your browser test.

  Request options are set at every test execution and apply to every step of your browser test at execution time, not recording time. 
  If you need these options to remain active to record the following steps, manually apply the options on the page you are recording from and create subsequent steps in your test.

> [1]: /synthetics/guide/identify_synthetics_bots/?tab=apitests

Reviewer: ```suggestion
[1]: /synthetics/guide/identify_synthetics_bots/?tab=apitests
[2]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
```

  {{% /tab %}}

  {{% tab "Certificate" %}}

  Select **Ignore server certificate error** to instruct the test to skip errors in the server certificate. 
  

>   * **Client Certificate**: If you have a private key or client certificate for the test recording, upload these files in your test by clicking **Upload File**. By default, only PEM certificates are accepted.

Reviewer: ```suggestion
  * **Client Certificate**: you can perform tests on systems requiring client certificates. To do so, simply upload the certificate file and the private key in your test by clicking **Upload File**. Only PEM certificates are accepted.
```

  * **Client Certificate Domains**: Once the certificate files are uploaded, the client certificate applies to the starting URL's domain. To apply the client certificate on another domain, specify the domain in the **Value** field.  

  You can include wildcards in the URL.

  {{% /tab %}}

  {{% tab "Proxy" %}}

  Enter a URL for a proxy you want to send requests through in the **Proxy URL** field as `http://<YOUR_USER>:<YOUR_PWD>@<YOUR_IP>:<YOUR_PORT>`.

  You can include [global variables](#use-global-variables) in the URL.

  {{% /tab %}}

  {{% tab "Privacy" %}}

  Select **Do not capture any screenshots for this test** to prevent screenshots from being taken in your test steps. 
  
  This privacy option is available as an [advanced option][1] at the individual test step level and ensures that no sensitive data appears in your test results. 
  

>   Preventing the test from taking step screenshots makes troubleshooting failures more difficult. For more information, see [Security][2].

Reviewer 2: ```suggestion
  Preventing the test from taking screenshots makes troubleshooting failures more difficult. For more information, see [Security][2].
```

  {{< /tabs >}}

3. **Name**: The name of your browser test.
4. **Select tags**: The `env` and related tags attached to your browser test. Use the `<KEY>:<VALUE>` format to filter on a `<VALUE>` for a given `<KEY>`.
5. **Browsers & Devices**: The browsers (`Chrome`, `Firefox`, and `Edge`) and devices (`Laptop Large`, `Tablet`, and `Mobile Small`) to run your test on. 
6. **Select locations**: The Datadog managed locations to run your test from. Many AWS locations from around the world are available for each site. You can also set up [private locations][1] to run your browser test from custom locations or from inside private networks. See a full list of locations in the [Datadog app][2] or use the [API][3]. {{< site-region region="gov" >}}**Note**: The West US (AWS GovCloud) location is supported on the Datadog for Government site.{{< /site-region >}}

> 7. **Select test frequency**: The intervals vary from every five minutes to once per week. [Contact Support][4] for one-minute  frequency.

Reviewer: let me check if this is really possible

Author: ```suggestion
7. **Select test frequency**: The intervals vary from every five minutes to once per week. [Contact Support][4] for one-minute frequency.
```

3. Specify a renotification frequency. To prevent renotification on failing tests, leave the option as `Never renotify if the monitor has not been resolved`.
4. Click **Save Details and Record Test**.

> ### Set permissions

Reviewer: can you please move the section ##permissions at the end of the file to this place?

Author: But this is the last step in the test creation workflow and we link to the ## Permissions below !

Reviewer: why not moving the section which is at the bottom to this place?

Author: imo one is calling out an action for users to set their permissions on the test vs. the broader "Permissions" description that we have in the paragraph about permissions. I'd also leave this step in to be consistent with Step 8 in the UI.

Restrict access to your browser test for yourself and custom roles by clicking the roles in the **Everyone in my org** dropdown menu. For more information, see [Permissions](#permissions).

## Record your steps

Tests can be only recorded from [Google Chrome][10]. To record your test, download the [Datadog Record Test extension for Google Chrome][11].

{{< img src="synthetics/browser_tests/browser_check_record_test2.png" alt="Browser test record test"  >}}

> 1. Optionally, select **Open in a pop-up** at the upper right of the page to open your test recording in a separate pop-up window. This is useful if your application does not support being opened in an iframe or if you want to avoid sizing issues at recording. You can also open the pop in **Incognito mode** to start recording your test from a fresh browser free of already logged-in sessions, cookies from your existing browser, and more.

Reviewer 2: ```suggestion
1. Optionally, select **Open in a pop-up** at the upper right of the page to open your test recording in a separate pop-up window. This is useful if your application does not support being opened in an iframe or if you want to avoid sizing issues at recording. You can also open the pop-up in **Incognito mode** to start recording your test from a fresh browser free of already logged-in sessions, cookies from your existing browser, and more.
```

If you have access to the [custom role feature][15], add your user to any custom role that includes `synthetics_read` and `synthetics_write` permissions.

> ### Select roles

Reviewer: ```suggestion
### Restrict access
```

> Restrict access to your browser test for yourself and custom roles by clicking the roles in the **Everyone in my org** dropdown menu.

Reviewer: ```suggestion
Restrict access to your browser test for yourself, everyone in your organization, or to specific roles by clicking the roles in the **Everyone in my org** dropdown menu. For more information about roles, see the [RBAC][14] documentation.
```

> Restrict access to your browser test for yourself and custom roles by clicking the roles in the **Everyone in my org** dropdown menu.

Reviewer: we should reuse the Restrict access already done for global var 
https://docs.datadoghq.com/synthetics/settings/?tab=specifyvalue#restrict-access

Author: Good point, I'll make a new screenshot!
